### PR TITLE
Handle UNUserNotificationCenter completionHandlers

### PR DESCRIPTION
**Pull request by blankg**
### Summary

Currently only didReceiveRemoteNotification: fetchCompletionHandler: are handle async by the JS complete method. This allows some time to perform tasks in background however the userNotificationCenter: didReceiveNotificationResponse: withCompletionHandler completion handler is called immediately which means not much time to perform tasks in background (about 1 sec from my experience) and calling complete from JS is meaningless.

This enhancement handles the NSUserNotificationCenter completionHandler in a similar way the  fetchCompletionHandler is handled.

### Checklist

- [ ] Supports `Android`
- [x ] Supports `iOS`
- [ ] `e2e` tests added or updated in [/tests/e2e/\*](/tests/e2e)
- [ ] Updated the documentation in the [docs repo](https://github.com/invertase/react-native-firebase-docs)
  - **LINK TO DOCS PR HERE**
- [ ] Flow types updated
- [ ] Typescript types updated

### Release Plan

[IOS][ENHANCEMENT] [NOTIFICATIONS] - handle NSUserNotificationCenter completion handlers

---

Think `react-native-firebase` is great? Please consider supporting the project with any of the below:

- 👉 Donate via [Open Collective](https://opencollective.com/react-native-firebase/donate)
- 👉 Follow [`React Native Firebase`](https://twitter.com/rnfirebase) and [`Invertase`](https://twitter.com/invertaseio) on Twitter
- 👉 Star this repo on GitHub ⭐️
- 👉 Contribute; see our [contributing guide](/CONTRIBUTING.md)

### CODE REVIEW

File: ios/RNFirebase/notifications/RNFirebaseNotifications.m [hunk 1]
```diff
@@ -214,9 +223,16 @@ - (void)userNotificationCenter:(UNUserNotificationCenter *)center
...
      NSDictionary *message = [self parseUNNotificationResponse:response];
+           
+     NSString *handlerKey = message[@"notification"][@"notificationId"];
 
      [self sendJSEvent:self name:NOTIFICATIONS_NOTIFICATION_OPENED body:message];
-     completionHandler();
+     if (handlerKey != nil) {
+         completionHandlers[handlerKey] = completionHandler;
+     }
+     else {
```
Comment (Salakar) [inline ios/RNFirebase/notifications/RNFirebaseNotifications.m]: Could you move this to the same line - `} else {` thanks

File: ios/RNFirebase/notifications/RNFirebaseNotifications.m [hunk 2]
```diff
@@ -100,10 +102,17 @@ - (void)didReceiveLocalNotification:(nonnull UILocalNotification *)localNotifica
...
-        void (^completionHandler)(UIBackgroundFetchResult) = completionHandlers[handlerKey];
-        if(completionHandler != nil) {
-            completionHandlers[handlerKey] = nil;
-            completionHandler(fetchResult);
+        void (^fetchCompletionHandler)(UIBackgroundFetchResult) = fetchCompletionHandlers[handlerKey];
+        if (fetchCompletionHandler != nil) {
+            fetchCompletionHandlers[handlerKey] = nil;
+            fetchCompletionHandler(fetchResult);
+        }
+        else {
```
Comment (Salakar) [inline ios/RNFirebase/notifications/RNFirebaseNotifications.m]: Could you move this to the same line - `} else {` thanks

File: ios/RNFirebase/notifications/RNFirebaseNotifications.m [hunk 3]
```diff
@@ -100,10 +102,17 @@ - (void)didReceiveLocalNotification:(nonnull UILocalNotification *)localNotifica
...
+            void(^completionHandler)(void) = completionHandlers[handlerKey];
```
Comment (Salakar) [inline ios/RNFirebase/notifications/RNFirebaseNotifications.m]: Any reason for removing the previous behaviour of using `UIBackgroundFetchResult`? Is this not going to break the previous behaviour? I might be getting a little confused on the changes here so apologies in advance 🙈

File: ios/RNFirebase/notifications/RNFirebaseNotifications.m [hunk 4]
```diff
@@ -100,10 +102,17 @@ - (void)didReceiveLocalNotification:(nonnull UILocalNotification *)localNotifica
...
+            if (completionHandler != nil) {
+                completionHandlers[handlerKey] = nil;
+                completionHandler();
```
Comment (Salakar) [inline ios/RNFirebase/notifications/RNFirebaseNotifications.m]: As above; any reason for removing the previous behaviour of `completionHandler(fetchResult);`? Is this not going to break the previous behaviour?

File: ios/RNFirebase/notifications/RNFirebaseNotifications.m (hunk 1 above)
Comment (blankg) [reply ios/RNFirebase/notifications/RNFirebaseNotifications.m]: will fix

File: ios/RNFirebase/notifications/RNFirebaseNotifications.m (hunk 2 above)
Comment (blankg) [reply ios/RNFirebase/notifications/RNFirebaseNotifications.m]: will fix

File: ios/RNFirebase/notifications/RNFirebaseNotifications.m (hunk 3 above)
Comment (blankg) [reply ios/RNFirebase/notifications/RNFirebaseNotifications.m]: The NSUserNotificationCenter completion handler doesn't take any arguments in its signature so you just need to call it.

File: ios/RNFirebase/notifications/RNFirebaseNotifications.m (hunk 4 above)
Comment (blankg) [reply ios/RNFirebase/notifications/RNFirebaseNotifications.m]: Same  as above the NSUserNotificationCenter completion handler doesn't take any arguments in its signature so you just need to call it.

**Comment by Salakar** [comment]
@blankg thanks for making those changes and replying to my points. It all makes sense now, sorry about the confusion 🙈

I'll go ahead and merge this now for the v5.2.0 release.

---
Loving `react-native-firebase` and the support we provide? Please consider supporting us with any of the below:

  - 👉  Back financially via [Open Collective](https://opencollective.com/react-native-firebase/donate)  
  - 👉  Follow [`React Native Firebase`](https://twitter.com/rnfirebase) and [`Invertase`](https://twitter.com/invertaseio) on Twitter 
  - 👉  Star this repo on GitHub ⭐️

**Pull request merged**